python/ray/includes/unique_ids.pxi

```diff
@@ -46,20 +46,32 @@ cdef extern from "ray/common/constants.h" nogil:
 
 cdef class BaseID:
 
+    @classmethod
+    def from_binary(cls, id_bytes):
+        return cls(id_bytes)
+
+    @classmethod
+    def from_hex(cls, hex_id):
+        raise NotImplementedError
+
     cdef size_t hash(self):
-        pass
+        raise NotImplementedError
 
     def binary(self):
-        pass
+        raise NotImplementedError
 
     def size(self):
-        pass
+        raise NotImplementedError
+
+    @staticmethod
+    def size():
+        raise NotImplementedError
```

Comment:
?

Reply:
we have these 2 size methods from cls and from instance.

Reply:
There is no need to have both class method and instance method. We can just keep the class method (not staticmethod) and class method can be called by an object instance as well